test-result-summary-client/src/Build/GitNewIssue.jsx

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import { params } from '../utils/query';
+import {  Button, Tooltip, Card } from 'antd';
+import TestBreadcrumb from './TestBreadcrumb';
+import { GithubOutlined } from '@ant-design/icons';
+import { getParams } from '../utils/query';
+import renderDuration from './Duration';
+
+import './table.css';
+
+import moment from 'moment';
+const DAY_FORMAT = 'MMM DD YYYY, hh:mm a';
+
+export default class GitNewissue extends Component {
+    state = {
+        urlParams: null,
+        firstFailJavaVersion: '',
+        firstFailDate: '',
+        firstFailUrl: '',
+        buildStartTime: '',
+        failMachine: [],
+    };
+
+    async componentDidMount() {
+        await this.updateData();
+    }
+
+    async updateData() {
+        const { testId, testName, duration, buildId, buildName, 
+            buildUrl, machine, buildTimeStamp, javaVersion, testResult } = getParams( this.props.location.search );
+            
+        let firstSeenFailed, firstFailDate, firstFailUrl, firstFailJavaVersion;
+        let failMachineMap = new Map();
+        let failMachineUrlBody = ``;
+        let failMachine = [];
+        
+        const fetchBuild = await fetch( `/api/getHistoryPerTest?testId=${testId}&limit=100`, {
+            method: 'get'
+        } );
+        const response = await fetchBuild.json();
+
+        // allFail has all failed tests
+        const allFail = response.filter(res => res.tests.testResult === 'FAILED');
```

Comment:
By filtering on `FAILED `, we will get incorrect info in this case. For example, if we have test history as the following:
F1, P2, P3, P4, F5, F6, F7 (F7 is the latest and F1 is the oldest), we would like to get F5 (the first seen failure). Using above code, we will lose all PASS state.